src/ctapipe/core/tool.py

```diff
@@ -440,11 +441,17 @@ def run(self, argv=None, raises=False):
                 )
                 exit_status = 130  # Script terminated by Control-C
             except Exception as err:
-                self.log.exception(f"Caught unexpected exception: {err}")
+                self.log.exception("Caught unexpected exception: %s", err)
                 Provenance().finish_activity(activity_name=self.name, status="error")
                 exit_status = 1  # any other error
                 if raises:
                     raise
+            except SystemExit as err:
+                exit_status = err.code
+                self.log.exception("Caught SystemExit with exit code %s", exit_status)
+                Provenance().finish_activity(
+                    activity_name=self.name, status="SystemExit"
```

Comment:
the original plan was to not allow any random exit code, only a pre-defined set.  If we need to have arbitrary exit codes (I guess the use case here is for workflow handling?), at the very least the code that was used should be stored in the provenance. Just recording "SystemExit" loses the reason for the exit happening. So we should either add a field to the provenance, or include the code in the status.  The latter is less ideal since it would require parsing, and means the sets of possible status values is not finite

Reply:
Yes, the use case here is workflow handling. And the exit codes should not be completely randomly, but rather controlled by the corresponding ICD in our case. However, given that `ctapipe` is a community product, I'm a bit hesitant to introduce such a dependency here. On the other hand, I fully agree with your point that the exit code should be stored in provenance. Here before making a move, I'd like to discuss options.

So far, the possible statuses are `["completed", "interrupted", "error"]` plus "SystemExit" I've added. I'm not sure what the subsequent purpose of this field. Is it just a human-readable label? If so, I would add another field, `exit_code` or just `code` where an actual exit code will be stored and change the list of possible statuses to `["success", "interrupted", "error", "partial_success"]` or similar. On the other hand, if the `status` field is supposed to be machine-readable, shall it just keep the exit code value?

Reply:
Perhaps, this should be default in the `_ActivityProvenance` helper class [constructor](https://github.com/cta-observatory/ctapipe/blob/2b702076effad4bcb2b9cd468b9784d7dc722ce2/src/ctapipe/core/provenance.py#L220-L230)?
```python
    def __init__(self, activity_name=sys.executable):
        self._prov = {
            "activity_name": activity_name,
            "activity_uuid": str(uuid.uuid4()),
            "start": {},
            "stop": {},
            "system": {},
            "input": [],
            "output": [],
        }
        self.name = activity_name
```

Reply:
> sys.executable

That's always the python interpreter, right? It won't be the cli tool name even if used via an entry point installed script

Reply:
this is the default value in the current code. It is overridden in the `Tool.run()` function:
https://github.com/cta-observatory/ctapipe/blob/2b702076effad4bcb2b9cd468b9784d7dc722ce2/src/ctapipe/core/tool.py#L409